kernel/src/engine/arrow_expression.rs

```diff
@@ -120,7 +120,33 @@ impl Scalar {
                         ArrowField::new(LIST_ARRAY_ROOT, t.element_type().try_into()?, true);
                     Arc::new(ListArray::new_null(Arc::new(field), num_rows))
                 }
-                DataType::Map { .. } => unimplemented!(),
+                DataType::Map(t) => {
+                    let MapType {
+                        key_type,
+                        value_type,
+                        .. // FIXME should use value_contains_null
+                    } = t.as_ref();
+                    if key_type != &DataType::STRING && value_type != &DataType::STRING {
+                        return Err(Error::generic("Only string key/values are supported"));
+                    }
+                    use arrow_array::builder::StringBuilder;
+                    let key_builder = StringBuilder::new();
+                    let val_builder = StringBuilder::new();
+                    let names = arrow_array::builder::MapFieldNames {
+                        entry: "entries".to_string(),
+                        key: "key".to_string(),
+                        value: "value".to_string(),
+                    };
+                    let mut builder = arrow_array::builder::MapBuilder::new(
+                        Some(names),
+                        key_builder,
+                        val_builder,
+                    );
+                    (0..num_rows).for_each(|_| {
+                        builder.append(true).unwrap();
```

Comment:
I think the top-level example in the docs also says:
```
// Construct `[{"joe": 1}, {"blogs": 2, "foo": 4}, {}, null]`
```
And we actually want the third case: `{}`.

I tried out `.append(false)` and it writes a top-level null instead of top-level empty map, so when we go to serialize `operationParameters`, it's gone (instead of our desired `operationParameters: {}`).

Reply:
come to think of it - maybe I've squeezed something incorrect into this API.. Null literal for the map would probably be what this API _should_ return. but the one we want is "empty MAP"..... hmmm this is unfortunate

Reply:
back to doing empty struct hack...